zebra/zebra_nhg.c

```diff
@@ -3635,7 +3635,13 @@ unsigned long zebra_nhg_score_proto(int type)
 		 * This should be the last ref if we remove client routes too,
 		 * and thus should remove and free them.
 		 */
-		zebra_nhg_decrement_ref(nhe);
+		if (!CHECK_FLAG(nhe->flags, NEXTHOP_GROUP_PROTO_RELEASED))
+			zebra_nhg_decrement_ref(nhe);
+		else {
+			if (IS_ZEBRA_DEBUG_RIB_DETAILED)
+				zlog_debug("%s: Refcnt-%d for nhe-%p with id %u already decremented in zread_nhg_del",
```

Comment:
can this be 
"%s: nhe %u (%p) refcount %u already decremented in zread_nhg_del"

Also add comment, /* protocol sends explicit delete of nhg, the refcount is decremented in zread_nhg_del() */

Reply:
done